File: app/src/main/java/com/github/rooneyandshadows/lightbulb/easyrecyclerviewdemo/demo/activity/MenuConfigurations.java
package com.github.rooneyandshadows.lightbulb.easyrecyclerviewdemo.demo.activity;

import android.view.View;

import com.github.rooneyandshadows.lightbulb.application.activity.BaseActivity;
import com.github.rooneyandshadows.lightbulb.application.activity.routing.BaseApplicationRouter;
import com.github.rooneyandshadows.lightbulb.application.activity.slidermenu.config.SliderMenuConfiguration;
import com.github.rooneyandshadows.lightbulb.application.activity.slidermenu.items.PrimaryMenuItem;
import com.github.rooneyandshadows.lightbulb.commons.utils.ResourceUtils;
import com.github.rooneyandshadows.lightbulb.easyrecyclerviewdemo.R;

import java.util.function.Consumer;

import kotlin.Unit;
import kotlin.jvm.functions.Function1;

public class MenuConfigurations {

    public static SliderMenuConfiguration getConfiguration(BaseActivity activity) {
        View headingView = activity.getLayoutInflater().inflate(R.layout.demo_drawer_header_view, null);
        SliderMenuConfiguration configuration = new SliderMenuConfiguration();
        configuration.withHeaderView(headingView);
        configuration.addMenuItem(new PrimaryMenuItem(
                -1,
                ResourceUtils.getPhrase(activity, R.string.regular_demo),
                null,
                null,
                1,
                action((slider) -> {
                    Router.getInstance().getRouter().toDemoRegular(BaseApplicationRouter.NavigationCommands.BACK_TO);
                    slider.closeSlider();
                })
        )).addMenuItem(new PrimaryMenuItem(
                -1,
                ResourceUtils.getPhrase(activity, R.string.selectable_demo),
                null,
                null,
                1,
                action((slider) -> {
                    Router.getInstance().getRouter().toDemoSelectable(BaseApplicationRouter.NavigationCommands.NAVIGATE_TO);
                    slider.closeSlider();
                })
        )).addMenuItem(new PrimaryMenuItem(
                -1,
                ResourceUtils.getPhrase(activity, R.string.header_and_footer_demo),
                null,
                null,
                1,
                action((slider) -> {
                    Router.getInstance().getRouter().toDemoHeaderAndFooter(BaseApplicationRouter.NavigationCommands.NAVIGATE_TO);
                    slider.closeSlider();
                })
        )).addMenuItem(new PrimaryMenuItem(
                -1,
                ResourceUtils.getPhrase(activity, R.string.empty_layout_demo),
                null,
                null,
                1,
                action((slider) -> {
                    Router.getInstance().getRouter().toEmptyLayout(BaseApplicationRouter.NavigationCommands.NAVIGATE_TO);
                    slider.closeSlider();
                })
        )).addMenuItem(new PrimaryMenuItem(
                -1,
                ResourceUtils.getPhrase(activity, R.string.bounce_effect_demo),
                null,
                null,
                1,
                action((slider) -> {
                    Router.getInstance().getRouter().toBounceEffect(BaseApplicationRouter.NavigationCommands.NAVIGATE_TO);
                    slider.closeSlider();
                })
        )).addMenuItem(new PrimaryMenuItem(
                -1,
                ResourceUtils.getPhrase(activity, R.string.pull_to_refresh_demo),
                null,
                null,
                1,
                action((slider) -> {
                    Router.getInstance().getRouter().toPullToRefresh(BaseApplicationRouter.NavigationCommands.NAVIGATE_TO);
                    slider.closeSlider();
                })
        )).addMenuItem(new PrimaryMenuItem(
                -1,
                ResourceUtils.getPhrase(activity, R.string.lazy_loading_demo),
                null,
                null,
                1,
                action((slider) -> {
                    Router.getInstance().getRouter().toLazyLoading(BaseApplicationRouter.NavigationCommands.NAVIGATE_TO);
                    slider.closeSlider();
                })
        )).addMenuItem(new PrimaryMenuItem(
                -1,
                ResourceUtils.getPhrase(activity, R.string.swipe_to_delete_demo),
                null,
                null,
                1,
                action((slider) -> {
                    Router.getInstance().getRouter().toSwipeToDelete(BaseApplicationRouter.NavigationCommands.NAVIGATE_TO);
                    slider.closeSlider();
                })
        )).addMenuItem(new PrimaryMenuItem(
                -1,
                ResourceUtils.getPhrase(activity, R.string.drag_to_reorder_demo),
                null,
                null,
                1,
                action((slider) -> {
                    Router.getInstance().getRouter().toDragToReorder(BaseApplicationRouter.NavigationCommands.NAVIGATE_TO);
                    slider.closeSlider();
                })
        )).addMenuItem(new PrimaryMenuItem(
                -1,
                ResourceUtils.getPhrase(activity, R.string.sticky_headers_simple_demo),
                null,
                null,
                1,
                action((slider) -> {
                    Router.getInstance().getRouter().toStickyHeadersSimple(BaseApplicationRouter.NavigationCommands.NAVIGATE_TO);
                    slider.closeSlider();
                })
        )).addMenuItem(new PrimaryMenuItem(
                -1,
                ResourceUtils.getPhrase(activity, R.string.sticky_headers_advanced_demo),
                null,
                null,
                1,
                action((slider) -> {
                    Router.getInstance().getRouter().toStickyHeadersAdvanced(BaseApplicationRouter.NavigationCommands.NAVIGATE_TO);
                    slider.closeSlider();
                })
        ));
        return configuration;
    }

    public static <T> Function1<T, Unit> action(Consumer<T> callable) {
        return t -> {
            callable.accept(t);
            return Unit.INSTANCE;
        };
    }
}

File: app/src/main/java/com/github/rooneyandshadows/lightbulb/easyrecyclerviewdemo/demo/activity/MenuConfigurations.kt
package com.github.rooneyandshadows.lightbulb.easyrecyclerviewdemo.demo.activity

import android.annotation.SuppressLint
import android.content.Context
import com.github.rooneyandshadows.lightbulb.application.activity.slidermenu.SliderMenu
import com.github.rooneyandshadows.lightbulb.application.activity.slidermenu.config.SliderMenuConfiguration
import com.github.rooneyandshadows.lightbulb.application.activity.slidermenu.config.SliderMenuConfiguration.HeaderConfiguration
import com.github.rooneyandshadows.lightbulb.application.activity.slidermenu.items.PrimaryMenuItem
import com.github.rooneyandshadows.lightbulb.commons.utils.ResourceUtils
import com.github.rooneyandshadows.lightbulb.easyrecyclerviewdemo.R
import com.github.rooneyandshadows.lightbulb.easyrecyclerviewdemo.lightbulb.service.LightbulbService.route

object MenuConfigurations {
    @SuppressLint("InflateParams")
    fun getConfiguration(context: Context): SliderMenuConfiguration {
        return SliderMenuConfiguration(HeaderConfiguration(R.layout.demo_drawer_header_view)).apply {
            itemsList.apply {
                add(
                    PrimaryMenuItem(
                        -1,
                        ResourceUtils.getPhrase(context, R.string.regular_demo),
                        null,
                        null,
                        1
                    ) { slider: SliderMenu ->
                        route().toDemoRegular().replace()
                        slider.closeSlider()
                    }
                )
                add(
                    PrimaryMenuItem(
                        -1,
                        ResourceUtils.getPhrase(context, R.string.filterable_demo),
                        null,
                        null,
                        1
                    ) { slider: SliderMenu ->
                        route().toDemoFilterable()
                            .forward()
                        slider.closeSlider()
                    }
                )
                add(
                    PrimaryMenuItem(
                        -1,
                        ResourceUtils.getPhrase(context, R.string.selectable_demo),
                        null,
                        null,
                        1
                    ) { slider: SliderMenu ->
                        route().toDemoSelectable()
                            .forward()
                        slider.closeSlider()
                    }
                )
                add(
                    PrimaryMenuItem(
                        -1,
                        ResourceUtils.getPhrase(context, R.string.flow_layout_manager_demo),
                        null,
                        null,
                        1
                    ) { slider: SliderMenu ->
                        route().toDemoFlowLayout()
                            .forward()
                        slider.closeSlider()
                    }
                )
                add(
                    PrimaryMenuItem(
                        -1,
                        ResourceUtils.getPhrase(context, R.string.header_and_footer_demo),
                        null,
                        null,
                        1
                    ) { slider: SliderMenu ->
                        route().toDemoHeaderAndFooter()
                            .forward()
                        slider.closeSlider()
                    }
                )
                add(
                    PrimaryMenuItem(
                        -1,
                        ResourceUtils.getPhrase(context, R.string.empty_layout_demo),
                        null,
                        null,
                        1
                    ) { slider: SliderMenu ->
                        route().toDemoEmptyLayout()
                            .forward()
                        slider.closeSlider()
                    }
                )
                add(
                    PrimaryMenuItem(
                        -1,
                        ResourceUtils.getPhrase(context, R.string.bounce_effect_demo),
                        null,
                        null,
                        1
                    ) { slider: SliderMenu ->
                        route().toDemoBouncy()
                            .forward()
                        slider.closeSlider()
                    }
                )
                add(
                    PrimaryMenuItem(
                        -1,
                        ResourceUtils.getPhrase(context, R.string.pull_to_refresh_demo),
                        null,
                        null,
                        1
                    ) { slider: SliderMenu ->
                        route().toDemoPullToRefresh()
                            .forward()
                        slider.closeSlider()
                    }
                )
                add(
                    PrimaryMenuItem(
                        -1,
                        ResourceUtils.getPhrase(context, R.string.lazy_loading_demo),
                        null,
                        null,
                        1
                    ) { slider: SliderMenu ->
                        route().toDemoLazyLoading()
                            .forward()
                        slider.closeSlider()
                    }
                )
                add(
                    PrimaryMenuItem(
                        -1,
                        ResourceUtils.getPhrase(context, R.string.swipe_to_delete_demo),
                        null,
                        null,
                        1
                    ) { slider: SliderMenu ->
                        route().toDemoSwipeToDelete()
                            .forward()
                        slider.closeSlider()
                    }
                )
                add(
                    PrimaryMenuItem(
                        -1,
                        ResourceUtils.getPhrase(context, R.string.drag_to_reorder_demo),
                        null,
                        null,
                        1
                    ) { slider: SliderMenu ->
                        route().toDemoDrag()
                            .forward()
                        slider.closeSlider()
                    }
                )
                add(
                    PrimaryMenuItem(
                        -1,
                        ResourceUtils.getPhrase(context, R.string.sticky_headers_simple_demo),
                        null,
                        null,
                        1
                    ) { slider: SliderMenu ->
                        route().toDemoStickyHeaders()
                            .forward()
                        slider.closeSlider()
                    }
                )
                add(
                    PrimaryMenuItem(
                        -1,
                        ResourceUtils.getPhrase(context, R.string.sticky_headers_advanced_demo),
                        null,
                        null,
                        1
                    ) { slider: SliderMenu ->
                        route().toDemoStickyHeadersAdvanced()
                            .forward()
                        slider.closeSlider()
                    }
                )
            }
        }
    }
}